msal_extensions/cache_lock.py

```diff
@@ -20,15 +22,19 @@ def __init__(self, lockfile_path):
             # More information here:
             # https://docs.python.org/3/library/fcntl.html#fcntl.lockf
             flags=portalocker.LOCK_EX | portalocker.LOCK_NB,
+            timeout=90,
             buffering=0)
 
     def __enter__(self):
+        self.start_time = str(time.time())
         file_handle = self._lock.__enter__()
         file_handle.write('{} {}'.format(os.getpid(), sys.argv[0]).encode('utf-8'))
         return file_handle
 
     def __exit__(self, *args):
         self._lock.__exit__(*args)
+        with open(self.time_interval_file_path, 'a+') as handle:
+            handle.write(self.start_time + "-" + str(time.time())+ "\n")
```

Comment:
[MSAL EX Java implemented it as part of the test suite](https://github.com/AzureAD/microsoft-authentication-extensions-for-java/commit/7d789a11f005dca07ad7b8c20060664f35ab4851#diff-62402c24972c337298ee12d0616b6aedR44). We should not modify our real `cache_lock.py` code path for this test.

[MSAL EX .Net did not implement this part](https://github.com/AzureAD/microsoft-authentication-extensions-for-dotnet/pull/82/files#diff-501fa80648fb3dcda4b2ce004dd24bebR45) at all. I think that is a reasonable move. Our offline conversation confirmed that in this PoC, the test failure could show up in either `validate_thread_locking_file()` or `validate_result_in_cache()`, so they seem the same problem, we do not have to test 2 files for the same underlying issue.

I would suggest to roll back the changes to this file, in order to have this PR eventually merged in.